algorithm.py

```diff
@@ -20,11 +24,8 @@
 
 Tutorial
 --------
-
-This module provides an abstraction for implementing arbitrary algorithms as a
-list of functions that operate on a shared state dictionary. Algorithms defined
-this way are easy to arbitrarily modify at run time, and they provide cascading
-exception handling.
+The algorithm model is a list of functions that are executed in
+defined order and update algorithm state stored in `state` dictionary.
```

Comment:
No. Algorithm is only linear, by design, because non-linear control flow is too complicated. The one exception is error handling, which already adds a complication I'm not entirely comfortable with.